### ISSUE TITLE
Basic auth for web server?

### ISSUE THREAD

**Issue by rubengargar** [datarhei/restreamer#383]
Hi,

I'm just updated our service to v2, and I see splendid features that I can use. One of them, is the integrated web server, but I would like to know if I can set some kind of Basic Auth for this custom web.

I have overwritten the generated index.html for my custom HTML, because I want some personalization further from default and also because I only need a big player view. Currently, I'm using Cloudflare->Nginx Reverse Proxy with Access List>Restreamer. Can I get rid of Nginx?

Thanks in advance!

**Comment by danielgoz** [comment]
@rubengargar - How exeactly are you config the Cloudflare->Nginx Reverse Proxy with Access List>Restreamer?
I am trying the same but can't figure out how.
Off course a native basic athorisation from the app would be way better.

**Comment by rubengargar** [comment]
Hi @danielgoz, we are moving away from this approach, but what we were doing is the following:
1. Under our **/data/www** of our Nginx Proxy Manager Docker we pasted the custom HTML&JS, based on the one you can find on this repo for the standard webplayer.
2. Create an **A subdomain** for our url on Cloudflare pointing to that NPM
3. On NPM, create an **Access List** for that web
4. On NPM, create a **Proxy Hos**t, with the domain name, destination ip and port (8080, 80...)
5. Check Cache Assets, Block Common Exploits. Select the previously created Access List
6. On **Custom Locations**, add the location with **both slash at start and end**, like "/location/". In forward hostname, we have to enter again the ip address with the same location, like "localhost/location/. Keep an eye on the slashes!
7. Add the SSL, and profit!

Some links about the configuration: https://github.com/NginxProxyManager/nginx-proxy-manager/issues/1027, https://github.com/NginxProxyManager/nginx-proxy-manager/issues/104

I don't know if you are using straight Nginx, in our case were using NPM for easier administration by our staff

**Comment by danielgoz** [comment]
Thanks man... I will give it a go...

**Comment by danielgoz** [comment]
Legend!
working as a charm.
it's kinda clanky though (sometimes it gets to a 403 forbbiden) but good enough for me.
the custom HTML&JS, thingy... is it because you want to customise the player page?
I dind't do anything like that...

**Comment by rubengargar** [comment]
Yep, that was only because we needed to show n number of players by each website, so we did a custom html page and the JS was for a dynamic list of players .
